logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java

```diff
@@ -309,11 +336,37 @@ private Dataset compile() {
             if (outputNodes.isEmpty()) {
                 return Dataset.IDENTITY;
             } else {
-                return DatasetCompiler.terminalDataset(outputNodes.stream().map(
+                return terminalDataset(outputNodes.stream().map(
                     leaf -> outputDataset(leaf, flatten(Collections.emptyList(), leaf))
                 ).collect(Collectors.toList()));
             }
         }
+        /**
```

Comment:
This is actually just moved code from `DatasetCompiler` here to keep consistency with the other `filterDataset`, `outputDataset`, ... methods.